asymmetric.go

```diff
@@ -430,13 +435,18 @@ func (ctx ecDecrypterSigner) signPayload(payload []byte, alg SignatureAlgorithm)
 		return Signature{}, err
 	}
 
+	keyBytes := curveBits / 8
+	if curveBits%8 > 0 {
+		keyBytes += 1
+	}
+
 	rBytes := r.Bytes()
-	rBytesPadded := make([]byte, keySize)
-	copy(rBytesPadded[keySize-len(rBytes):], rBytes)
+	rBytesPadded := make([]byte, keyBytes)
+	copy(rBytesPadded[keyBytes-len(rBytes):], rBytes)
```

Comment:
A comment explaining how this padding works would be nice (that `r.Bytes()` is big-endian and needs zeroes prepended to make it `keyBytes` long).